src/abstracts.ts

```diff
@@ -10,6 +10,70 @@ export interface SB3ProjectJSON {
     targets: SpriteProperties[];
 }
 
+// Serialized Block Interfaces - We should maybe move the serialization typedefs some other file?
+export enum SB3SerializedInputType {
+    INPUT_SAME_BLOCK_SHADOW = 1,
+    INPUT_BLOCK_NO_SHADOW = 2,
+    INPUT_DIFF_BLOCK_SHADOW = 3,
+}
+
+export enum SB3SerializedBlockType {
+    MATH_NUM_PRIMITIVE = 4,
+    POSITIVE_NUM_PRIMITIVE = 5,
+    WHOLE_NUM_PRIMITIVE = 6,
+    INTEGER_NUM_PRIMITIVE = 7,
+    ANGLE_NUM_PRIMITIVE = 8,
+    COLOR_PICKER_PRIMITIVE = 9,
+    TEXT_PRIMITIVE = 10,
+    BROADCAST_PRIMITIVE = 11,
+    VAR_PRIMITIVE = 12,
+    LIST_PRIMITIVE = 13,
+}
+
+export enum ScratchVariableTypes {
+    SCALAR_TYPE = '',
+    LIST_TYPE = 'list',
+    BROADCAST_MESSAGE_TYPE = 'broadcast_msg',
+}
+
+export type SB3SerializedField = [any, string?];
+
+export interface SB3SerializedFields {
+    [name: string]: SB3SerializedField;
```

Comment:
Does the key [name: string] mean that the key can be any string, or is it just 'name'?

Reply:
any string